op-devstack/dsl/fb_ws_proxy.go

```diff
@@ -41,29 +42,53 @@ func (c *FlashblocksWebsocketProxy) Escape() stack.FlashblocksWebsocketProxy {
 }
 
 func (c *FlashblocksWebsocketProxy) ListenFor(logger log.Logger, duration time.Duration, output chan<- []byte, done chan<- struct{}) error {
-	return websocketListenFor(logger, c.Escape().WsUrl(), duration, output, done)
+	wsURL := c.Escape().WsUrl()
+	headers := c.Escape().WsHeaders()
+	return websocketListenFor(logger, wsURL, headers, duration, output, done)
 }
 
-func websocketListenFor(logger log.Logger, wsURL string, duration time.Duration, output chan<- []byte, done chan<- struct{}) error {
+func websocketListenFor(logger log.Logger, wsURL string, headers http.Header, duration time.Duration, output chan<- []byte, done chan<- struct{}) error {
 	defer close(done)
-	logger.Debug("Testing WebSocket connection to", "url", wsURL)
+	logger.Debug("Testing WebSocket connection to", "url", wsURL, "headers", headers)
+
+	// Log the headers for debug purposes
+	if headers != nil {
+		for key, values := range headers {
+			logger.Debug("Header", "key", key, "values", values)
+		}
+	} else {
+		logger.Debug("No headers provided")
+	}
 
 	dialer := &websocket.Dialer{
 		HandshakeTimeout: 6 * time.Second,
 	}
 
-	conn, _, err := dialer.Dial(wsURL, nil)
+	logger.Debug("Attempting WebSocket connection", "url", wsURL)
+	conn, resp, err := dialer.Dial(wsURL, headers)
 	if err != nil {
+		logger.Error("WebSocket connection failed", "url", wsURL, "error", err)
+		if resp != nil {
+			logger.Error("HTTP response details", "status", resp.Status, "headers", resp.Header)
+			resp.Body.Close()
+		}
 		return fmt.Errorf("failed to connect to Flashblocks WebSocket endpoint %s: %w", wsURL, err)
 	}
+
+	// Always close the response body to prevent resource leaks
```

Comment:
Nitpick: you can move this below the .Dial line and it will cover all cases, err or not